### ISSUE TITLE
vue/no-mutating-props not flagging `.sync` usage

### ISSUE THREAD

**Issue by douglasg14b** [vuejs/eslint-plugin-vue#1714]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** `6.7.2`
- **eslint-plugin-vue version:** : `7.19.1`
- **Node version:** : `14` (I think?)
- **Operating System:** : `Windows 10`

**Please show your full configuration:**

```json5
module.exports = {
  root: true,
  parserOptions: {
    parser: 'babel-eslint'
  },
  env: {
    browser: true,
    node: true,
    'jest/globals': true
  },
  extends: [
    'standard',
    'plugin:vue/recommended',
    'plugin:vuetify/base',
    'plugin:vuetify/recommended'
  ],
  plugins: [
    'vue',
    'jest',
    'import'
  ],
  settings: {
    'import/resolver': {
      alias: {
        map: [
          ['@', './src']
        ],
        extensions: ['.vue', '.js', '.gql']
      }
    }
  },
  rules: {
    'import/extensions': [
      'error',
      {
        'vue': 'always',
        'mixin': 'never',
        'js': 'never',
        'gql': 'always'
      }
    ],
    'no-console': 'off',
    'no-debugger': 'warn',
    'no-return-await': 'off',

    'vue/v-slot-style': [
      'warn',
      {
        atComponent: 'v-slot',
        default: 'longform',
        named: 'longform'
      }
    ],

    'vue/html-closing-bracket-newline': [
      'error',
      {
        singleline: 'never',
        multiline: 'never'
      }
    ],
    'vue/html-self-closing': [
      'error',
      {
        html: {
          void: 'always',
          normal: 'never',
          component: 'never'
        },
        svg: 'always',
        math: 'always'
      }
    ]
  },
  globals: {}
}

```

**What did you do?**	

```vue
<template>
    <account-details-form :customer-number.sync="customerNumber"></account-details-form>
</template>

<script>
import AccountDetailsForm from '@/components/Accounts/AccountDetailsForm.vue'

export default {
  name: 'AccountDetailsDialog',
  components: {
    AccountDetailsForm
  },
  props: {
    customerNumber: {
      type: [Number, String],
      required: true
    }
  },
  data () {
    return {
      isDialogOpen: false,
      form: {}
    }
  }
}
</script>

```

**What did you expect to happen?**

That eslint would flag the `:customer-number.sync="customerNumber"` as an error for mutating a prop

**What actually happened?**

It does not

**Repository to reproduce this issue**

The repo seems unnecessary, eslint isn't breaking.
